oracle/orchestrator/lifecycle.go

```diff
@@ -0,0 +1,119 @@
+package orchestrator
+
+import (
+	"context"
+	"fmt"
+
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
+)
+
+// CtxErrors is a map of context errors that we check for when starting the provider.
+// We only want cancel the main error group if the context is canceled or the deadline
+// is exceeded. Otherwise failures should be handled gracefully.
+var CtxErrors = map[error]struct{}{
+	context.Canceled:         {},
+	context.DeadlineExceeded: {},
+}
+
+// GeneralProvider is a interface for a provider that implements the base provider.
+type GeneralProvider interface {
+	// Start starts the provider.
+	Start(ctx context.Context) error
+	// Stop stops the provider.
+	Name() string
+}
```

Comment:
I think this interface should be defined in `base` provider dir.  

And we should also add a type assertion that all of our providers implement this interface.

Reply:
I was thinking we can probably just migrate most of the functionality on the base provider into the provider types directory as additional interfaces. Ill make this a non-global interface for the time being.